pkg/urbit/jets/e/pedersen.c

```diff
@@ -0,0 +1,100 @@
+
+/* j/5/pedersen.c
+**
+*/
+#include "all.h"
+#include <urcrypt.h>
+
+static c3_y*
+u3r_bytes_all_works(c3_w* len_w, u3_atom a)
+{
+  c3_w met_w;
+  if (a == u3_nul) {
+    met_w = *len_w = 1;
+  } else {
+    met_w = *len_w = u3r_met(3, a);
+  }
+  return u3r_bytes_alloc(0, met_w, a);
+}
+
+static c3_y *
+fold_hash(u3_atom x, c3_w *len_acc) {
+  c3_w len_x, pos = 0, ext_pos = 0;
+  c3_y *acc = NULL, *out = NULL, *ext = NULL;
+  c3_y *hed, *dat_x = u3r_bytes_all_works(&len_x, x);
+  c3_w ext_len = (len_x / 64) + 1;
+  bool pol = true; // polarity - true=byte aligned, false=shift 4 bits
+  while (1) {
+    hed = (dat_x + pos);
+    if (pol && (len_x - pos) >= 32) {
+      // if positive polarity and chunk is the full 252 bits then
+      // zero out the final 4 bits but save them in *ext to hash
+      // at the very end
+      if ((hed[31] & 0xf0) != 0) {
+        if (ext == NULL) {
+          ext = u3a_malloc(ext_len);
+        }
+        ext[ext_pos] = (hed[31] & 0xf0) >> 4;
+        ext_pos++;
+        hed[31] = hed[31] & 0x0f;
+      }
+    }
+    if (pos == 0) {
```

Comment:
very minor style issue, but please put extra spaces in the conditionals, and put the constants first in the comparisons, e.g.

    if ( 0 == pos_w ) {